### ISSUE TITLE
sincerely request to public the pre-trained model for PCN dataset

### ISSUE THREAD

**Issue by songqikong** [CuiRuikai/Partial2Complete#13]
amazing job in self-supervised pc completion area.
but i can't get the counterapart cdl2 score in the paper for PCN dataset when i train the model with vanilla setup.

so I sincerely request the author to release the pre-trained model of the PCN dataset.

**Comment by CuiRuikai** [comment]
Thank you for your interest on our work.
The vanilla setting is not suitable for every category, so I did make some adjustments on the weights for different categories.
Unfortunately, I don’t have access to the PC that stores the weights for at least the next two months.
I am confident that everyone should be able to reproduce the reported scores.  
Please let me know any problems you encountered, so I can help you reproduce the results.

**Comment by songqikong** [comment]
> Thank you for your interest on our work. The vanilla setting is not suitable for every category, so I did make some adjustments on the weights for different categories. Unfortunately, I don’t have access to the PC that stores the weights for at least the next two months. I am confident that everyone should be able to reproduce the reported scores. Please let me know any problems you encountered, so I can help you reproduce the results.

Appreciate your timely reply! That really dissolve my confusion.

**Comment by CuiRuikai** [comment]
Before you comment any specific questions, I would like to share some empirical tricks to run reproduce the results for every category in PCN.
1. The weight for NCC loss should be reduced if the geometry is usually complex or not flat in a category, e.g. lamp, watercraft.
2. Make the weight of loss for reconstruction loss low if the result lack completion capability.
3. To be noted, samples in the PCN dataset is much more incomplete than 3D-EPN, so we are training with complete samples from 3D-EPN as we clarified in the paper.

**Comment by songqikong** [comment]
> Before you comment any specific questions, I would like to share some empirical tricks to run reproduce the results for every category in PCN.
> 
> 1. The weight for NCC loss should be reduced if the geometry is usually complex or not flat in a category, e.g. lamp, watercraft.
> 2. Make the weight of loss for reconstruction loss low if the result lack completion capability. For example, 1 for recon loss, 1000 for completion loss.
> 3. To be noted, samples in the PCN dataset is much more incomplete than 3D-EPN, so we are training with complete samples from 3D-EPN as we clarified in the paper. To train with complete samples, it is easy by copying all the complete samples to the same folder of partial samples. And if you want to sample complete data more frequently than partial samples, you can just replicate the complete samples.

Hello again! I tried train EPN dataset these days. Stay, cant obtain the right cd score. So, it is my problem or the configyaml in your code is not correct?

**Comment by CuiRuikai** [comment]
May I confirm which setting you are referring to? a. purely train the P2C on 3D-EPN dataset; b. jointly train PCN with EPN? 
I have provided the pre-category weights for the 3D-EPN dataset. Even we use the same config, the results should be close to the reported results.

**Comment by songqikong** [comment]
> May I confirm which setting you are referring to? a. purely train the P2C on 3D-EPN dataset; b. jointly train PCN with EPN? I have provided the pre-category weights for the 3D-EPN dataset. Even we use the same config, the results should be close to the reported results.

NoNoNo, this time i just use the config file in the path "cfgs/EPN3D-models/P2C.yaml" train EPN dataset.
so you mean using this very config file can wrok for every single taxonomy in EPN dataset, right?
